GliaWidgets/SecureConversations/SecureConversations.ChatWithTranscriptModel.swift

```diff
@@ -429,26 +437,116 @@ extension SecureConversations {
             action?(.sendButtonHidden(!isValid))
             return isValid
         }
+    }
+}
 
-        private func sendMessage() {
-            guard validateMessage() else { return }
-
-           _ = environment.sendSecureMessage(
-                messageText,
-                fileUploadListModel.attachment,
-                environment.queueIds
-           ) { [weak self] result in
-               switch result {
-               case .success:
-                   self?.messageText = ""
-                   self?.fileUploadListModel.removeSucceededUploads()
-                   // TODO: MOB-1738
-               case let .failure(error):
-                   // TODO: MOB-1874
-                   break
-               }
+// MARK: Message management
+extension SecureConversations.TranscriptModel {
+    private func sendMessage() {
+        guard validateMessage() else { return }
+
+        let uploads = fileUploadListModel.succeededUploads
+        let localFiles = uploads.map(\.localFile)
+
+        let outgoingMessage = OutgoingMessage(
+            content: messageText,
+            files: localFiles
+        )
+        appendItem(.init(kind: .outgoingMessage(outgoingMessage)), to: messagesSection, animated: true)
+
+       _ = environment.sendSecureMessage(
+            messageText,
+            fileUploadListModel.attachment,
+            environment.queueIds
+       ) { [weak self] result in
+           guard let self = self else { return }
+           switch result {
+           case let .success(message):
+               self.replace(
+                outgoingMessage,
+                uploads: uploads,
+                with: message,
+                in: self.messagesSection
+               )
+           case let .failure(error):
+               // TODO: MOB-1874
+               break
            }
-        }
+       }
+
+        // Clear inputs, thus disabling them
+        // and preventing the message to be sent again.
+        messageText = ""
```

Comment:
NIT: maybe these cleanup actions can be its own small method. Not super necessary, but it seems to me here like the send has already several parts: the validation, the preparation (of files and stuff), the sending, and the cleanup.

Reply:
Moved it into separate method 6564e2daf2894759c00f133f1b98524324e3c750.